crates/biome_js_analyze/src/lint/nursery/use_consistent_member_accessibility.rs

```diff
@@ -0,0 +1,558 @@
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub enum Accessibility {
+    #[default]
+    NoPublic,
+    Explicit,
+    None,
+}
+
+impl Rule for UseConsistentMemberAccessibility {
+    type Query = Ast<AnyMember>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = UseConsistentMemberAccessibilityOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+
+        match ctx.options().accessibility {
+            Accessibility::NoPublic => {
+                if node.is_no_public() == Some(false) {
+                    Some(())
+                } else {
+                    None
+                }
+            }
+            Accessibility::Explicit => {
+                if node.is_explicit() == Some(false) {
+                    Some(())
+                } else {
+                    None
+                }
+            }
+            Accessibility::None => {
+                if node.is_none() == Some(false) {
+                    Some(())
+                } else {
+                    None
+                }
+            }
+        }
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        match ctx.options().accessibility {
+            Accessibility::NoPublic => Some(
+                RuleDiagnostic::new(
+                    rule_category!(),
+                    node.property_range(),
+                    markup! {
+                        "Missing accessibility modifier on member."
+                    },
+                )
+                .note(markup! {
+                    "Public accessibility modifier cannot be used."
+                }),
+            ),
+            Accessibility::Explicit => Some(
+                RuleDiagnostic::new(
+                    rule_category!(),
+                    node.syntax().text_trimmed_range(),
+                    markup! {
+                        "Missing accessibility modifier on member."
+                    },
+                )
+                .note(markup! {
+                    "Accessibility modifier is required."
+                }),
+            ),
+            Accessibility::None => Some(
+                RuleDiagnostic::new(
+                    rule_category!(),
+                    node.property_range(),
+                    markup! {
+                        "Missing accessibility modifier on member."
+                    },
+                )
+                .note(markup! {
+                    "Accessibility modifier cannot be used."
+                }),
+            ),
+        }
+    }
+}
+
+declare_node_union! {
+    pub AnyMember = JsConstructorClassMember | JsPropertyClassMember | JsMethodClassMember | TsPropertyParameter | JsGetterClassMember | JsSetterClassMember
```

Comment:
We should also cover:
- signature class members (i.e. member without body, i.e. abstract methods and properties)
- property parameters (for example `constructor(readonly prop) {}` where an accessibility is missing)

Also, I suggest renaming the node to `AnyJsMemberWithAccessibility`

```suggestion
    pub AnyJsMemberWithAccessibility = JsConstructorClassMember | JsPropertyClassMember | JsMethodClassMember | TsPropertyParameter | JsGetterClassMember | JsSetterClassMember
    | TsConstructorSignatureClassMember | TsMethodSignatureClassMember | TsPropertySignatureClassMember | TsGetterSignatureClassMember | TsSetterSignatureClassMember
	| TsPropertyParameter
```

To find the existing nodes, you can take a look at `xtask/codegen/js.ungramar`, we have `AnyJsClassMember`